### ISSUE TITLE
eval_linear.py for resnet50: accuracy too low compared to eval logs

### ISSUE THREAD

**Issue by mgpadalkar** [facebookresearch/dino#131]
I tried the [eval_linear.py](https://github.com/facebookresearch/dino/blob/main/eval_linear.py) script for resnet50 using the dino pretrained weights.
I don't know if I did something wrong but I see the accuracy too low compared to the [eval logs](https://dl.fbaipublicfiles.com/dino/dino_resnet50_pretrain/dino_resnet50_pretrain_eval_linear_log.txt) which has something like this:
```
{"train_lr": 0.2999999999999728, "train_loss": 2.0051621776301185, "test_loss": 1.3246705343046457, "test_acc1": 68.338, "test_acc5": 89.05, "epoch": 0}
{"train_lr": 0.2999259840548763, "train_loss": 1.4219337276347748, "test_loss": 1.2068134845827547, "test_acc1": 70.606, "test_acc5": 90.31, "epoch": 1}
{"train_lr": 0.2997040092642101, "train_loss": 1.314717579769207, "test_loss": 1.1542176046334873, "test_acc1": 71.474, "test_acc5": 90.934, "epoch": 2}
```

Instead what I get is:
```
{"train_lr": 0.0019999999999998686, "train_loss": 6.739332184237192, "epoch": 0, "test_loss": 6.532769427007558, "test_acc1": 30.232000001220705, "test_acc5": 55.100000008544924}
{"train_lr": 0.0019995065603657376, "train_loss": 6.403291306017496, "epoch": 1, "test_loss": 6.166384765080044, "test_acc1": 38.762000017089846, "test_acc5": 66.66400000488281}
{"train_lr": 0.0019980267284282105, "train_loss": 6.078300359865403, "epoch": 2, "test_loss": 5.812116973254145, "test_acc1": 41.66000000366211, "test_acc5": 70.29800001220703}
```

Not only is the accuracy ~38% instead of ~70%, the loss is also ~5x higher.

With a single GPU the command I used is:
```
python eval_linear.py --data_path imagenet1k_folder --num_workers 4 --arch resnet50 --batch_size_per_gpu 256
```
Nevertheless, the accuracy is as expected with the [pretrained head](https://dl.fbaipublicfiles.com/dino/dino_resnet50_pretrain/dino_resnet50_linearweights.pth).

My environment is the suggested one (python version 3.6, PyTorch version 1.7.1, CUDA 11.0 and torchvision 0.8.2). I get similar results with python=3.7, pytorch=1.8.1, cuda=10.2 and torchvision=0.9.1.

Looking at [args](https://dl.fbaipublicfiles.com/dino/dino_resnet50_pretrain/args.txt), it feels like `--lr 0.3` should also be passed to `eval_linear.py` for resnet50. I will try it out to see if this issue is resolved. 

Will appreciate if you could point to something I may not be doing correct.

**Comment by woctezuma** [comment]
Relevant thread:
- #84

**Comment by mgpadalkar** [comment]
Thank you @woctezuma!

> I use a learning rate of 0.3 and a total batch size of 256 (2 gpus, batch size per gpu is 128). #84 (comment) :heavy_check_mark: ***very relevant***

I confirm `--lr 0.3` did the trick! Below is the new eval log having expected values.
```
{"train_lr": 0.599999999999982, "train_loss": 2.0123278156929, "epoch": 0, "test_loss": 1.323740845736192, "test_acc1": 68.378, "test_acc5": 89.00200002441406}
{"train_lr": 0.5998519681097447, "train_loss": 1.4198951296838722, "epoch": 1, "test_loss": 1.2045549640850144, "test_acc1": 70.50800003417969, "test_acc5": 90.31799998779297}
{"train_lr": 0.5994080185284917, "train_loss": 1.3115100268121247, "epoch": 2, "test_loss": 1.1510053140168288, "test_acc1": 71.67199999511719, "test_acc5": 91.08200002929688}
```

> There must be something to tweak regarding learning rate or batch size.
> Hopefully, with the right value for `lr`, this works! :heavy_check_mark: 

I also confirm it is only affected by `lr` and not by the batch size. On a machine with 4 GPUs and with the following command,
```
python -m torch.distributed.launch --nnodes=1 --nproc_per_node=4 --use_env eval_linear.py --data_path imagenet1k_folder --num_workers 20 --arch resnet50 --batch_size_per_gpu 768 --lr 0.3
```
I again got the expected values!
```
{"train_lr": 3.599999999999982, "train_loss": 2.0603999546127456, "epoch": 0, "test_loss": 1.350174142555757, "test_acc1": 68.052002109375, "test_acc5": 88.90000234375}
{"train_lr": 3.599111808658331, "train_loss": 1.4234871527082043, "epoch": 1, "test_loss": 1.2270691205154767, "test_acc1": 70.30200205078125, "test_acc5": 90.2000026953125}
{"train_lr": 3.596448111170898, "train_loss": 1.3121466108343818, "epoch": 2, "test_loss": 1.188134934414517, "test_acc1": 70.96600173828125, "test_acc5": 90.7340026171875}
{"train_lr": 3.592011536285553, "train_loss": 1.249356233902525, "epoch": 3, "test_loss": 1.1379844266356844, "test_acc1": 72.43400240234375, "test_acc5": 91.17600234375}
{"train_lr": 3.5858064623660275, "train_loss": 1.2040533952450638, "epoch": 4, "test_loss": 1.1160629129771031, "test_acc1": 72.7920021875, "test_acc5": 91.4620027734375}
```

**Comment by mgpadalkar** [comment]
In the end after 100 epochs, my numbers are similar to https://github.com/facebookresearch/dino/issues/84#issuecomment-884237015
> A fresh clone and that command results in 75.034% which is much closer.
